### PR TITLE
PathDB API extended to bound number of results returned. Small PS fixes.

### PR DESCRIPTION
**Pull request by pszal**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/546%23issuecomment-160615807%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23issuecomment-160641867%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23discussion_r46138039%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23discussion_r46138285%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23discussion_r46144657%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23discussion_r46144742%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23issuecomment-160615807%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40kormat%20%22%2C%20%22created_at%22%3A%20%222015-11-30T12%3A25%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-11-30T14%3A14%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20d163104df7cc563f704526de357bc6a46c4edea3%20lib/path_db.py%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23discussion_r46138285%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Lower%20fidelity%20is%20better%3F%22%2C%20%22created_at%22%3A%20%222015-11-30T12%3A43%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22yes%2C%20as%20long%20as%20fidelity%20is%20defined%20as%20path%20length.%22%2C%20%22created_at%22%3A%20%222015-11-30T13%3A56%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/path_db.py%3AL280-288%22%7D%2C%20%22Pull%20d163104df7cc563f704526de357bc6a46c4edea3%20infrastructure/path_server.py%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/546%23discussion_r46138039%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Might%20be%20worth%20just%20adding%202%20methods%20to%20PathSegment%2C%20to%20return%20the%20src/dst%20isd%20ads.%22%2C%20%22created_at%22%3A%20%222015-11-30T12%3A40%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20think%20it%20makes%20sense.%20I%27ll%20update%20that%20with%20a%20next%20PR.%22%2C%20%22created_at%22%3A%20%222015-11-30T13%3A55%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/path_server.py%3AL409-417%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/kormat%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/kormat'></a>
- [ ] <a href='#crh-comment-Pull d163104df7cc563f704526de357bc6a46c4edea3 infrastructure/path_server.py 29'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/546#discussion_r46138039'>File: infrastructure/path_server.py:L409-417</a></b>
- <a href='https://github.com/kormat'></a> Might be worth just adding 2 methods to PathSegment, to return the src/dst isd ads.
- <a href='https://github.com/pszalach'></a> I think it makes sense. I'll update that with a next PR.
- [ ] <a href='#crh-comment-Pull d163104df7cc563f704526de357bc6a46c4edea3 lib/path_db.py 45'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/546#discussion_r46138285'>File: lib/path_db.py:L280-288</a></b>
- <a href='https://github.com/kormat'></a> Lower fidelity is better?
- <a href='https://github.com/pszalach'></a> yes, as long as fidelity is defined as path length.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/546#issuecomment-160615807'>General Comment</a></b>
- <a href='https://github.com/pszalach'></a> @kormat

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/546?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/546?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/546?approve=0'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/546'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: infrastructure/path_server.py [hunk 1]
```diff
@@ -408,10 +409,8 @@ def _handle_down_segment_record(self, pkt):
         paths_to_propagate = []
         paths_to_master = []
         for pcb in records.pcbs[PST.DOWN] + records.pcbs[PST.UP_DOWN]:
-            src_isd = pcb.get_first_pcbm().isd_id
-            src_ad = pcb.get_first_pcbm().ad_id
-            dst_ad = pcb.get_last_pcbm().ad_id
-            dst_isd = pcb.get_last_pcbm().isd_id
+            src_isd, src_ad = pcb.get_first_pcbm().get_isd_ad()
```
Comment (kormat) [inline infrastructure/path_server.py:412]: Might be worth just adding 2 methods to PathSegment, to return the src/dst isd ads.

File: lib/path_db.py [hunk 2]
```diff
@@ -276,6 +280,8 @@ def __call__(self, *args, **kwargs):
             self._db.delete(expired_recs)
         pcbs = sorted([r['record'] for r in valid_recs],
                       key=lambda x: x.fidelity)
+        if self._max_res_no and not full:
+            pcbs = pcbs[:self._max_res_no]
```
Comment (kormat) [inline lib/path_db.py:284]: Lower fidelity is better?

File: infrastructure/path_server.py (hunk 1 above)
Comment (pszal) [inline infrastructure/path_server.py:412]: I think it makes sense. I'll update that with a next PR.

File: lib/path_db.py (hunk 2 above)
Comment (pszal) [inline lib/path_db.py:284]: yes, as long as fidelity is defined as path length.

**Pull request merged**